hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/debug/KeyPathRetriever.java

```diff
@@ -0,0 +1,295 @@
+    }
+    // out stream
+    PrintWriter writer;
+    if (StringUtils.isEmpty(fileName)) {
+      writer = out();
+    } else {
+      writer = new PrintWriter(new BufferedWriter(new PrintWriter(fileName, UTF_8.name())));
+    }
+
+    // db handler
+    OMMetadataManager metadataManager = getOmMetadataManager(dbFile);
+    if (metadataManager == null) {
+      writer.close();
+      return null;
+    }
+
+    dirTreeDbStore = openDb(new File(dbFile).getParentFile());
+    if (dirTreeDbStore == null) {
+      writer.close();
+      metadataManager.stop();
+      return null;
+    }
+    dirTreeTable = dirTreeDbStore.getTable(DIRTREE_COLUMN_FAMILY.getName(), Long.class, String.class);
+
+    try {
+      retrieve(metadataManager, writer, containerIds);
+    } finally {
+      dirTreeDbStore.close();
+      writer.close();
+      metadataManager.stop();
+    }
+    return null;
+  }
+
+  public void retrieve(
+      OMMetadataManager metadataManager, PrintWriter writer, Set<Long> containerIds) {
+    // build dir tree
+    Map<Long, Pair<Long, String>> bucketVolMap = new HashMap<>();
+    try {
+      prepareDirIdTree(metadataManager, bucketVolMap);
+    } catch (Exception e) {
+      System.out.println("Exception occurred reading directory Table, " + e);
+      return;
+    }
+
+    // iterate file table and filter for container
+    try (TableIterator<String, ? extends KeyValue<String, OmKeyInfo>> fileItr
+             = metadataManager.getFileTable().iterator()) {
+      while (fileItr.hasNext()) {
+        KeyValue<String, OmKeyInfo> next = fileItr.next();
+        boolean found = next.getValue().getKeyLocationVersions().stream().anyMatch(
+            e -> e.getLocationList().stream().anyMatch(
+                blk -> containerIds.contains(blk.getBlockID().getContainerID())));
+        if (found) {
+          StringBuilder sb = new StringBuilder(next.getValue().getKeyName());
+          Long prvParent = next.getValue().getParentObjectID();
+          while (prvParent != null) {
+            // check reached for bucket volume level
+            if (bucketVolMap.containsKey(prvParent)) {
+              Pair<Long, String> nameParentPair = bucketVolMap.get(prvParent);
+              sb.insert(0, nameParentPair.getValue() + OM_KEY_PREFIX);
+              prvParent = nameParentPair.getKey();
+              if (null == prvParent) {
+                // add to output as reached till volume
+                writer.println(sb);
+                break;
+              }
+              continue;
+            }
+
+            // check dir tree
+            Pair<Long, String> nameParentPair = getFromDirTree(prvParent);
+            if (nameParentPair == null) {
+              break;
+            }
+            sb.insert(0, nameParentPair.getValue() + OM_KEY_PREFIX);
+            prvParent = nameParentPair.getKey();
+          }
+        }
+      }
+    } catch (Exception e) {
+      System.out.println("Exception occurred reading file Table, " + e);
```

Comment:
Errors can be printed in err instead of out, out can be just used for actual output.